codex-rs/cli/src/mcp_cmd.rs

```diff
@@ -201,15 +204,33 @@ fn run_list(config_overrides: &CliConfigOverrides, list_args: ListArgs) -> Resul
         let json_entries: Vec<_> = entries
             .into_iter()
             .map(|(name, cfg)| {
-                let env = cfg.env.as_ref().map(|env| {
-                    env.iter()
-                        .map(|(k, v)| (k.clone(), v.clone()))
-                        .collect::<BTreeMap<_, _>>()
-                });
+                let env = match &cfg.transport {
+                    McpServerTransportConfig::Stdio { env, .. } => env.as_ref().map(|env| {
+                        env.iter()
```

Comment:
Why is this not a straight clone? Does it have to be a `BTreeMap`? Isn't there a `.from()` or `into()` you can use instead here?

Reply:
This is just indented from the original impl (view the PR without whitespace). I didn't want to make other modifications here